guava-tests/test/com/google/common/collect/TopKSelectorTest.java

```diff
@@ -119,4 +121,37 @@ public int compare(Integer o1, Integer o2) {
     assertThat(top.topK()).containsExactlyElementsIn(Collections.nCopies(k, 0));
     assertThat(compareCalls[0]).isAtMost(10L * n * IntMath.log2(k, RoundingMode.CEILING));
   }
+
+  public void testExceedMaxIteration() {
```

Comment:
Is there a chance we can find a single, small, fixed input which would have previously triggered the bug?  It should be deterministic, no?

Reply:
Maybe. I wonder if using a property-based testing framework, like [jqwik](https://jqwik.net/), help us find it? I ask because they have the ability to "shrink" offending inputs to the smallest input that reproduces a discovered bug. :thinking:

Reply:
We can eliminate any flakiness by finding a seed to `Random` that reliably reproduces the value -- like `1629833645599L`, for example :)

It would be even better to have a small and understandable input, but I'm very happy for the bug report and fix regardless!